src/pages/index.js

```diff
@@ -0,0 +1,393 @@
+const previewModalCaptionEl = previewModal.querySelector('.modal__caption');
+const previewModalCloseBtn = previewModal.querySelector('.modal__close-btn');
+
+//Card related elements
+const cardTemplate = document.querySelector('#card-template');
+const cardsList = document.querySelector('.cards__list');
+
+// select the avatar modal and its elements
+const avatarModal = document.querySelector('#avatar-modal');
+const avatarForm = avatarModal.querySelector('.modal__form');
+const avatarSubmitBtn = avatarModal.querySelector('.modal__submit-btn');
+const avatarModalCloseBtn = avatarModal.querySelector('.modal__close-btn');
+const avatarInput = avatarModal.querySelector('#profile-avatar-input');
+
+// Delete Modal elements
+const deleteModal = document.querySelector('#delete-modal');
+const deleteForm = deleteModal.querySelector('.modal__form');
+const deleteModalCancelBtn = deleteModal.querySelector('#delete-cancel-btn');
+let selectedCard;
+let selectedCardId;
+
+function openModal(modal) {
+  modal.classList.add('modal_opened');
+  document.addEventListener('keydown', handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove('modal_opened');
+  document.removeEventListener('keydown', handleEscapeKey);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  // editModalSubmitBtn.textContent = 'Saving...';
+  // change text content to "Saving..."
+  const editModalSubmitBtn = evt.submitter;
+  editModalSubmitBtn.textContent = 'Saving...';
+
+  setButtonText(editModalSubmitBtn, true, "Save", "Saving...");
+  // disable the button
+  api.editUserInfo({ name: editModalNameInput.value, about: editModalDescriptionInput.value })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+
+      // todo - call setButtonText instead
+      editModalSubmitBtn.textContent = "Save";
+    });
+}
+
+// todo - implement loading text for all other form submissions
+
+function handleAddCardFormSubmit(evt) {
+  evt.preventDefault();
+
+  const addCardModalSubmitBtn = evt.submitter;
+  addCardModalSubmitBtn.textContent = 'Saving...';
+  setButtonText(addCardModalSubmitBtn, true, "Save", "Saving...");
+
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      console.log(data);
+
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+    }).catch(console.error)
+    .finally(() => {
+
+      addCardModalSubmitBtn.textContent = "Save";
+    });
+  disableButton(cardSubmitBtn, settings);
+  closeModal(cardModal);
+
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const avatarSubmitBtn = evt.submitter;
+
+  avatarSubmitBtn.textContent = 'Saving...';
```

Comment:
` avatarSubmitBtn.textContent = 'Saving...';` is not needed here because it's done inside `setButtonText`